third_party/move/move-compiler-v2/tests/checking/typing/freeze_mut_ref.move

```diff
@@ -0,0 +1,52 @@
+module 0x42::freeze_mut_ref {
+    use std::vector;
+
+    public fun borrow_mut<Element>(
+        map: &mut vector<Element>,
+    ): &Element {
+        vector::borrow_mut(map, 0)
+    }
+
+    public fun borrow_mut2<Element>(
+        v: &mut Element,
+    ): &Element {
+        v
+    }
+
+    public fun borrow_mut3<Element>(
+        v1: &mut Element,
+        v2: & Element
+    ): &Element {
+        if (true)
+            v1
+        else
+            v2
+    }
+
+    public fun borrow_mut4<Element>(
+        v: &mut Element,
+    ): &Element {
+        return v
+    }
+
+    struct S {
+
+    }
+
+    fun t0() {
+        let x: &u64 = &mut 0; x;
+    }
+
+    fun t1(s: &mut S): &S {
+        s
+    }
+
+    fun t2(u1: &mut u64, u2: &mut u64): (&u64, &mut u64) {
+        (u1, u2)
+    }
+
+    fun t3(s: &mut S) {
+        *(s: &S) = S {  };
+    }
+
```

Comment:
There may be a few cases that are not resolved and still have this issue. An example:

```
    public fun t4() {
        let x: &u64;
        let y: &u64;
        (x, y) = (&mut 0, &mut 0);
    }
```

If you add this as a transactional test, you will see it leads to a bytecode verification error.

Reply:
Note that we do not run the final codegen and thus bytecode verification for tests under `checking`. Since this is fixing an error that manifested as a bytecode verification error, it would be good to add some of these tests in transactional tests as well.

Reply:
Thanks for catching it, @vineethk! There is a missing case for checking assignments.

Reply:
Added a transactional test.

Reply:
Interesting.  What's different between this case and https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-compiler/transactional-tests/tests/constants/by_reference.move and the same test for v2?  `&mut BYTES` is allowed, somehow.

Reply:
Do you mean `let b = &mut BYTES;`?  Here `b` is implicitly has the type `&mut` while `x` and `y` in the example above has the type `&`.

Reply:
> Note that we do not run the final codegen and thus bytecode verification for tests under `checking`. Since this is fixing an error that manifested as a bytecode verification error, it would be good to add some of these tests in transactional tests as well.

Maybe we should run through all passes for all checking tests.  It will take more time, but eng time is more expensive.